### DISCUSSION TITLE
We should enable `nodelay` for RDP.

### DISCUSSION THREAD

**Discussion by fernvenue** [rathole-org/rathole#193] [Show and tell]
Hi, I'm using rathole for a while, that works so great. And I just tested performance for RDP when `nodelay` enabled and disabled. 

- When `nodelay` is enabled, FPS can be 30 or even more, and complex changes to the display will not cause feedback delays.
- When `nodelay` is disabled, the FPS behaves fine at the beginning, but as we start doing some things, the FPS gradually drops below 10 and gradually approaches 3 or even 2 or 1. Importantly, complex display introduce significant feedback delays that accumulate over time. Even sometimes you will have to re-establish an RDP connection.

By the way, TCP no delay doesn't introduce new problems in general, so let's enable it! Of course, this is just my personal opinion, and my simple test is not universal, just sharing a little personal experience :)

**Comment by rawvoid** [comment]
Hi, @fernvenue 

How to display the RDP FPS value?

Thanks!

**Reply by fernvenue** [reply]
RDP doesn't have this feature, but you can use any other tool to do that when you're in the full screen, such as Xbox Game Bar, NVIDIA GeForce Experience and so on.

**Reply by rawvoid** [reply]
> RDP doesn't have this feature, but you can use any other tool to do that when you're in the full screen, such as Xbox Game Bar, NVIDIA GeForce Experience and so on.

Thank you, a simple way is to open the https://testufo.com/ website to monitor the fps of rdp.

**Comment by fernvenue** [comment]
Glad to see that `nodelay` enabled by default since [v0.4.7](https://github.com/rapiz1/rathole/releases/tag/v0.4.7).
